### PR TITLE
Update FPGA READMEs for 3rd party board support

### PR DESCRIPTION
**Pull request by akertesz**
## Description

The FPGA code samples in the Tutorials directory all now support 3rd party/custom boards. All of the READMEs must be updated to reflect this. I've updated all of the READMEs to reflect this.

As part of this change, I also clarify the supported OSes, and updated the DevCloud compilation instructions.

You can see the rendered README on my fork:
https://github.com/akertesz/oneAPI-samples/blob/README-custom-board/DirectProgramming/DPC++FPGA/Tutorials/Features/kernel_args_restrict/README.md

## Type of change

- [X] README update

## How Has This Been Tested?

Check the rendered README for formatting errors.

### CODE REVIEW

**Review by mdbtucker — APPROVED** [review]
Looks good to me, thanks Audrey.

File: DirectProgramming/DPC++FPGA/Tutorials/Features/kernel_args_restrict/README.md [hunk 1]
```diff
@@ -8,14 +8,13 @@ The [oneAPI Programming Guide](https://software.intel.com/en-us/oneapi-programmi
 
 | Optimized for                     | Description
 ---                                 |---
-| OS                                | Linux* Ubuntu* 18.04; Windows* 10
-| Hardware                          | Intel® Programmable Acceleration Card (PAC) with Intel Arria® 10 GX FPGA; <br> Intel® FPGA Programmable Acceleration Card (PAC) D5005 (with Intel Stratix® 10 SX)
+| OS                                | Linux Ubuntu 18.04/20.04; RHEL/CentOS 8; SUSE 15; Windows 10
```
Comment (mdbtucker) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/kernel_args_restrict/README.md:11]: Did you intend to delete the '*' from all the OS names, or was that an oversight?  I believe the '*' is still used elsewhere in the file, so this is at minimum inconsistent.

File: DirectProgramming/DPC++FPGA/Tutorials/Features/kernel_args_restrict/README.md [hunk 2]
```diff
@@ -130,7 +135,10 @@ When compiling for FPGA hardware, it is recommended to increase the job timeout
      ```
      nmake report
      ``` 
-   * An FPGA hardware target is not provided on Windows*. 
+   * Compile for FPGA hardware (longer compile time, targets FPGA device):
+     ```
+     nmake fpga
+     ``` 
 
 *Note:* The Intel® PAC with Intel Arria® 10 GX FPGA and Intel® FPGA PAC D5005 (with Intel Stratix® 10 SX) do not yet support Windows*. Compiling to FPGA hardware on Windows* requires a third-party or custom Board Support Package (BSP) with Windows* support.
```
Comment (mdbtucker) [inline DirectProgramming/DPC++FPGA/Tutorials/Features/kernel_args_restrict/README.md:143]: Should we get rid of the word 'yet' here, while we're at it?  I'm not aware of any plans to ever support Windows with these cards.

File: DirectProgramming/DPC++FPGA/Tutorials/Features/kernel_args_restrict/README.md (hunk 1 above)
Comment (akertesz) [reply DirectProgramming/DPC++FPGA/Tutorials/Features/kernel_args_restrict/README.md]: Heh, I assume those italics are a reference to the missing * s.
It was looking awfully busy so I removed them, but that was a mistake. Fixed in the latest commit.

File: DirectProgramming/DPC++FPGA/Tutorials/Features/kernel_args_restrict/README.md (hunk 2 above)
Comment (akertesz) [reply DirectProgramming/DPC++FPGA/Tutorials/Features/kernel_args_restrict/README.md]: Me neither. I've removed the "yet".

**Review by JoeOster — DISMISSED** [review]

**Review by mdbtucker — DISMISSED** [review]

**Comment by akertesz** [comment]
Please don't merge yet -- I'm waiting on Aparna to give me a thumbs up. Then I'll propagate the change to the other DPC++FPGA/Tutorials READMEs using this same PR.

**Comment by JoeOster** [comment]
> Please don't merge yet -- I'm waiting on Aparna to give me a thumbs up. Then I'll propagate the change to the other DPC++FPGA/Tutorials READMEs using this same PR.

Either retitle it to "WIP" or mark as a draft

**Review by mdbtucker — APPROVED** [review]
Thanks for all the 'grunt work' updates Audrey.

**Pull request merged**